### PR TITLE
add CI for Bouffalolab BL602 

### PR DESCRIPTION
**Pull request by stoneM2017**
#### Problem
No CI for BL602 platform.

#### Change overview
add CI for BL602 to build lighting-app examples.

#### Testing
test using self created docker image and pass the "Build example - BL602" job.

### CODE REVIEW

**Comment by stoneM2017** [comment]
@woody-apple @andy31415 Hi, we want to add CI for Bouffalolab bl602 platfrom, could you kindly help to review this PR to see whether it can be merged or not, although there are 5 failed checks. Thanks a lot!
I checked these 5 failed checks, all seems not caused by this PR:
1. Build example - BL602 / BL602 (pull_request):  seems we need your help to push the docker image to docker hub, the docker image for bl602 is already built successfully.
2. other 4 failures: related to docker image building failures in other platforms.

File: integrations/docker/images/chip-build-bl602/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ARG VERSION=latest
+FROM connectedhomeip/chip-build:${VERSION}
+
+RUN set -x \
+    && apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -fy --no-install-recommends \
+    git=1:2.25.1-1ubuntu3.4 \
```
Comment (andy31415) [inline integrations/docker/images/chip-build-bl602/Dockerfile:7]: This seems very close to just chip-build. Why is a separate dockerfile needed?

Comment (stoneM2017) [reply integrations/docker/images/chip-build-bl602/Dockerfile:7]: Thanks for your review, we see there are different dockerfile and docker build/image for different platfrom, and considering the further extensions to integrate more bl602 specific tools/scripts, so we decide to use a separate dockerfile.

File: integrations/docker/images/chip-build-bl602/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ARG VERSION=latest
```
[the remark this answers is not in the document]
Comment (stoneM2017) [reply integrations/docker/images/chip-build-bl602/Dockerfile:1]: Thanks for your review, so what we need to do?

File: .github/workflows/examples-bl602.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Set up environment for size reports
+        if: ${{ !env.ACT }}
+        env:
+          GH_CONTEXT: ${{ toJson(github) }}
+        run: scripts/tools/memory/gh_sizes_environment.py "${GH_CONTEXT}"
+
+      - name: Build example BL602 Lighting App
+        timeout-minutes: 30
+        run: |
+          scripts/examples/gn_bl602_example.sh lighting-app ./out/bl602-light
```
[the remark this answers is not in the document]
Comment (stoneM2017) [reply .github/workflows/examples-bl602.yaml:58]: Yes, build_examples.py can be used to build lighting-app. 
Currently, gn_bl602_example.sh is still needed to build lighting-app with pw_rpc enabled, we will take in mind to not add more platform specific shell scripts and retire the gn_bl602_example.sh in the future.

File: integrations/docker/images/chip-build-bl602/Dockerfile (hunk 1 above)
Comment (stoneM2017) [reply integrations/docker/images/chip-build-bl602/Dockerfile]: changed to use chip-build.

File: .github/workflows/examples-bl602.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (stoneM2017) [reply .github/workflows/examples-bl602.yaml]: already changed to use build_examples.py to build the lighting-app example.

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by stoneM2017** [comment]
> @stoneM2017 - the CI for BL602 seems to easily run out of space. I tried to find maintainers for BL602 in slack but had no answers. Who maintains this platform and how can we reach out for support?
> 
> We need some level of support/responsiveness to keep this platform as a maintained CI target.

@andy31415 you can reach me for support. 

For the "run out of space" issue, do you mean the follow error:
95.95obj/third_party/connectedhomeip/third_party/bouffalolab/bl602_sdk/repo/components/platform/hosal/bl602_hal/sdk.hosal_ota.c.o: in function `hosal_ota_start':
/__w/connectedhomeip/connectedhomeip/out/bl602-light-rpc/../../examples/lighting-app/bouffalolab/bl602/third_party/connectedhomeip/third_party/bouffalolab/bl602_sdk/repo/components/platform/hosal/bl602_hal/hosal_ota.c:87:(.text.hosal_ota_start+0x3c): relocation truncated to fit: R_RISCV_JAL against symbol `memset' defined in .text section in /__w/connectedhomeip/connectedhomeip/third_party/bouffalolab/bl602_sdk/repo/toolchain/riscv/Linux/bin/../lib/gcc/riscv64-unknown-elf/8.3.0/../../../../riscv64-unknown-elf/lib/rv32imf/ilp32f/libc.a(lib_a-memset.o)
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.

We will take a look and fix ASAP.

**Pull request merged**